tests/BaGetter.Core.Tests/Metadata/DefaultPackageMetadataServiceTests.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using NuGet.Versioning;
+using Xunit;
+
+namespace BaGetter.Core.Tests.Metadata;
+
+public class DefaultPackageMetadataServiceTests
+{
+    private readonly Mock<IUrlGenerator> _urlGenerator;
+    private readonly RegistrationBuilder _registrationBuilder;
+
+    public DefaultPackageMetadataServiceTests()
+    {
+        _urlGenerator = new Mock<IUrlGenerator>();
+        _registrationBuilder = new RegistrationBuilder(_urlGenerator.Object);
+    }
+
+    [Fact]
+    public void Ctor_PackageServiceIsNull_ShouldThrow()
+    {
+        // Arrange
+        var urlGenerator = new Mock<IUrlGenerator>();
+        var registrationBuilder = new RegistrationBuilder(urlGenerator.Object);
```

Comment:
What about the readonly fields? If this is the only test needing these objects, the fields can be removed.